### PR TITLE
Add ModuleSetupFooter component.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #1045 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Great work, @asvinb. Looks good to me in general. Added a few minor comments with some suggestions. Please, take a look.

File: assets/js/components/setup/ModuleSetupFooter.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { Cell, Grid, Row } from '../../material-components';
+import Link from '../Link';
+const { useSelect } = Data;
+
+export default function ModuleSetupFooter( { module, onCancelButtonClick } ) {
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/setup/ModuleSetupFooter.js]: Sure thing! Updated!

File: assets/js/components/setup/ModuleSetupFooter.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		select( module?.storeName )?.isSetupBlocked?.()
+	);
+
+	if ( ! module ) {
+		return null;
+	}
+
+	return (
+		<div className="googlesitekit-setup__footer">
+			<Grid>
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/setup/ModuleSetupFooter.js:52]: Unfortunately there are some existing CSS rules which will break the layout and require some further tweaks to the CSS. I suggest we leave it like that, unless you are very keen on removing that extra wrapper :) Let me know

File: assets/js/components/setup/ModuleSetupFooter.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+					</Cell>
+				</Row>
+			</Grid>
+		</div>
+	);
+}
+
+ModuleSetupFooter.propTypes = {
+	module: PropTypes.shape( {
+		slug: PropTypes.string.isRequired,
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/setup/ModuleSetupFooter.js:73]: Good catch!

File: assets/js/components/setup/ModuleSetupFooter.test.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+/**
+ * ModuleSetupFooter component tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/setup/ModuleSetupFooter.test.js]: 👍

File: assets/js/components/setup/ModuleSetupFooter.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/**
+ * ModuleSetupFooter component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
[the remark this answers is not in the document]
Comment (asvinb) [reply assets/js/components/setup/ModuleSetupFooter.js]: 👍

File: assets/js/components/setup/ModuleSetupFooter.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/components/setup/ModuleSetupFooter.js:52]: Ok, that's fine, just wondered if we could combine those :smile: Thanks, @asvinb.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @asvinb. Looks good to me. Added one more suggestion that I'll apply by myself and then I'll approve it.

**Review by eugene-manuilov — APPROVED** [review]

File: assets/js/components/setup/ModuleSetupFooter.test.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		[ 'Cancel', 'is', false ],
+		[ 'Back', 'is not', true ],
+	] )(
+		'should render the %s label when setup %s blocked',
+		( label, _, isBlocked ) => {
+			validateIsSetupBlockedError = isBlocked;
+			const { container } = render(
+				<ModuleSetupFooter
+					module={ module }
+					onCancelButtonClick={ () => {} }
```
Comment (eugene-manuilov) [inline assets/js/components/setup/ModuleSetupFooter.test.js:71]: Ok, this one also need to be updated: 
```suggestion
					onCancel={ () => {} }
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Found one more occurrence of the `onCancelButtonClick` property. I will update it.

**Review by eugene-manuilov — APPROVED** [review]
All good now.

**Pull request merged**